### PR TITLE
Kafka producer and consumer, checkstyle

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

This PR adds base for our `test-clients`:
1. kafka producer and consumer
2. checkstyle and suppressions
3. image building
4. travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  directories:
+    - "$HOME/.m2"
+env:
+  global:
+    - PULL_REQUEST=${TRAVIS_PULL_REQUEST}
+    - COMMIT=${TRAVIS_COMMIT::8}
+    - TAG=${TRAVIS_TAG:-latest}
+    - BRANCH=${TRAVIS_BRANCH:-main}
+    - DOCKER_REGISTRY=quay.io
+    - DOCKER_ORG=strimzi-examples
```
Comment (scholzj) [inline .travis.yml:19]: This should have a separate repo.

File: kafka/consumer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM centos:7
```
Comment (scholzj) [inline kafka/consumer/Dockerfile:1]: Use UBI8

File: kafka/consumer/Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+PROJECT_NAME=kafka-consumer
```
Comment (scholzj) [inline kafka/consumer/Makefile:1]: I think you should rename this to make things less confusing.

File: kafka/consumer/pom.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>strimzi.io</groupId>
+        <artifactId>kafka-test-clients</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>kafka-consumer</artifactId>
```
Comment (scholzj) [inline kafka/consumer/pom.xml:12]: Rename to make it less confusing?

File: kafka/consumer/pom.xml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>strimzi.io</groupId>
```
Comment (scholzj) [inline kafka/consumer/pom.xml:8]: `io.strimzi`?

File: kafka/pom.xml [hunk 6]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>strimzi.io</groupId>
```
Comment (scholzj) [inline kafka/pom.xml:8]: `io.strimzi`?

File: kafka/producer/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,16 @@
+FROM centos:7
```
Comment (scholzj) [inline kafka/producer/Dockerfile:1]: Use UBI8.

File: kafka/producer/Makefile [hunk 8]
```diff
@@ -0,0 +1,11 @@
+PROJECT_NAME=kafka-producer
```
Comment (scholzj) [inline kafka/producer/Makefile:1]: This should maybe have some better name? `test-producer`?

**Review by scholzj — COMMENTED** [review]
Do you really need a structure with so many levels?

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  directories:
+    - "$HOME/.m2"
+env:
+  global:
+    - PULL_REQUEST=${TRAVIS_PULL_REQUEST}
+    - COMMIT=${TRAVIS_COMMIT::8}
+    - TAG=${TRAVIS_TAG:-latest}
+    - BRANCH=${TRAVIS_BRANCH:-main}
+    - DOCKER_REGISTRY=quay.io
+    - DOCKER_ORG=strimzi
```
Comment (scholzj) [inline .travis.yml:19]: This should be different org?

Comment (Frawless) [reply .travis.yml:19]: like `strimzi-test-client` ?

Comment (scholzj) [reply .travis.yml:19]: That would seem like a good org for this.

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Comment by im-konge** [comment]
`Do you really need a structure with so many levels?`
> How we discussed offline, we want to add HTTP clients (and maybe more in the future) to this repo, so from my POV the structure is correct.

**Pull request merged**